### PR TITLE
creating new radius and tacacs data classes 

### PR DESCRIPTION
**Pull request by nikhilkp93**
# Pull Request summary:
creating new radius and tacacs data classes.
We would need couple of new options based on whether user wants to select vpn directly or if they want to select source VPN along with VXLAN tunnel.

# Description of changes:
- When we use direct VPN0/VPN512, and use vpn,vpn subnet ip, it will fail. it only needs source vpn, which is not there.

- On the other hand, when we use VPN 512 via VXLAN, then sourcevpn, vpn, vpn subnet ip are requited. Which is also not there.

- Hence we need new data classes

# Checklist:
- [X] Make sure to run pre-commit before committing changes
- [X] Make sure all checks have passed
- [X] PR description is clear and comprehensive
- [X] Mentioned the issue that this PR solves (if applicable)
- [X] Make sure you test the changes

### CODE REVIEW

File: catalystwan/dataclasses.py [hunk 1]
```diff
@@ -453,8 +453,9 @@ class RadiusServer(DataclassBase):
     address: str = field(metadata={FIELD_NAME: "address"})
     auth_port: int = field(metadata={FIELD_NAME: "authPort"})
     acct_port: int = field(metadata={FIELD_NAME: "acctPort"})
-    vpn: int = field(metadata={FIELD_NAME: "vpn"})
-    vpn_ip_subnet: str = field(metadata={FIELD_NAME: "vpnIpSubnet"})
+    sourceVpn: Optional[int] = field(metadata={FIELD_NAME: "sourceVpn"})
```
Comment (renukavinay) [inline catalystwan/dataclasses.py:456]: create a subclass and update the new fields in the subclass. donot change existing characteristics.
Also, these payloads should be backward compatible.

**Review by renukavinay — COMMENTED** [review]
Please create a subclass for this new fields instead of changing properties

File: catalystwan/dataclasses.py (hunk 1 above)
Comment (nikhilkp93) [reply catalystwan/dataclasses.py]: made the change, please re-review

File: catalystwan/dataclasses.py [hunk 2]
```diff
@@ -460,6 +460,18 @@ class RadiusServer(DataclassBase):
...
 
 
+@define(frozen=True)
+class ExtendedRadiusServer(RadiusServer):
+    """
+    Extended RADIUS server with additional fields.
+    """
+
+    sourceVpn: Optional[int] = field(metadata={FIELD_NAME: "sourceVpn"})
+    vpn: Optional[int] = field(metadata={FIELD_NAME: "vpn"})
```
Comment (sbasan) [inline catalystwan/dataclasses.py:470]: `vpn` field already defined in base class: `RadiusServer`

File: catalystwan/dataclasses.py [hunk 3]
```diff
@@ -460,6 +460,18 @@ class RadiusServer(DataclassBase):
     priority: int = field(metadata={FIELD_NAME: "priority"})
 
 
+@define(frozen=True)
+class ExtendedRadiusServer(RadiusServer):
+    """
+    Extended RADIUS server with additional fields.
+    """
+
+    sourceVpn: Optional[int] = field(metadata={FIELD_NAME: "sourceVpn"})
```
Comment (sbasan) [inline catalystwan/dataclasses.py:469]: please use snake case as identifier: `source_vpn`

**Review by renukavinay — APPROVED** [review]
lgtm

**Review by smialy — APPROVED** [review]

**Review by sbasan — APPROVED** [review]

**Pull request merged**